linera-views/src/key_value_store_view.rs

```diff
@@ -91,15 +110,17 @@ where
         let keys = vec![key1, key2];
         let values_bytes = context.read_multi_key_bytes(keys).await?;
         let hash = from_bytes_opt(&values_bytes[0])?;
-        let key_sizes = match &values_bytes[1] {
-            None => BTreeMap::new(),
+        let key_sizes_vec = match &values_bytes[1] {
+            None => Vec::new(),
             Some(bytes) => bcs::from_bytes(bytes)?,
         };
+        let key_sizes_map = convert_key_sizes_vec(&key_sizes_vec);
         Ok(Self {
             context,
             was_cleared: false,
             updates: BTreeMap::new(),
-            key_sizes,
+            key_sizes_vec,
```

Comment:
Do we need that field? Otherwise we wouldn't have to clone all the keys when converting to a map.

Reply:
We need it in order to have the rollback functionality.